ASM/c/pots.c

```diff
@@ -0,0 +1,66 @@
+#include "pots.h"
+#include "n64.h"
+#include "gfx.h"
+
+#define POT_SIDE_TEXTURE 0x06000000
+#define POT_DLIST 0x060017C0
+
+uint32_t POT_TEXTURE_MATCH_CONTENTS = 0;
+
+extern void *GILDED_CHEST_FRONT_TEXTURE;
+extern void *SILVER_CHEST_FRONT_TEXTURE;
+extern void *SKULL_CHEST_FRONT_TEXTURE;
+
+extern Mtx_t *write_matrix_stack_top(z64_gfx_t *gfx);
+asm(".equ write_matrix_stack_top, 0x800AB900");
+
+override_t get_pot_override(z64_actor_t *actor, z64_game_t *game) {
+    //make a dummy EnItem00 with enough info to get the override
+    EnItem00 dummy;
+    dummy.collectibleFlag = (actor->variable & 0x7E00) >> 8;
+    dummy.actor.actor_id = 0x15;
+
+    return lookup_override(&dummy, game->scene_index, 0);
+}
+
+void draw_pot(z64_actor_t *actor, z64_game_t *game) {
+    z64_gfx_t *gfx = game->common.gfx;
+    Gfx *opa_ptr = gfx->poly_opa.p;
+
+    //write matrix
+    Mtx_t *mtx = write_matrix_stack_top(gfx);
+
+    //get chest_type
+    override_t override = get_pot_override(actor, game);
+    uint16_t resolved_item_id = resolve_upgrades(override.value.item_id);
```

Comment:
There appears to be an exception thrown in resolve_upgrades, causing a crash.

Reply:
looks like the only way resolve_upgrades would throw an exception would be if get_item_row returns NULL. Check what value of item_id is getting passed in.

Reply:
also you should set dummy.actor.dropFlag = 1;